### PR TITLE
Fix image feature fp16 overflow in Molmo2

### LINKED ISSUE #1163: Molmo2: float16 vision tower overflow produces all-NaN logits (output: !!!!!!!!)
## Summary

When running `mlx-community/Molmo2-8B-4bit` (and likely other 4-bit Molmo2 variants), generation outputs an unbroken stream of `!` characters regardless of prompt or image. The root cause is a float16 overflow in the vision tower that corrupts all language model logits.

## Reproduction

```python
import mlx_vlm

model, processor = mlx_vlm.load("mlx-community/Molmo2-8B-4bit", trust_remote_code=True)
config = model.config
formatted = mlx_vlm.apply_chat_template(processor, config, "Describe what you see.", num_images=1)
result = mlx_vlm.generate(
    model, processor, formatted,
    image=["path/to/any_image.png"],
    max_tokens=200, temperature=0.5, verbose=True,
)
# Output: !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! (200 tokens of token ID 0)
```

Also reproducible via the CLI:
```
python -m mlx_vlm generate --model mlx-community/Molmo2-8B-4bit \
  --prompt "Describe what you see." --image path/to/image.png \
  --trust-remote-code --max-tokens 200
```

## Root cause

The 4-bit Molmo2 vision tower produces **2 feature values that exceed the float16 maximum (~65504)**, causing them to become `Inf`. These `Inf` values are merged into the text embeddings in `get_input_embeddings`. The subsequent `RMSNorm` in block 0 then computes `Inf / Inf = NaN`, corrupting all 151,936 logits. `argmax` of an all-NaN array returns index 0, which in Molmo's tokenizer decodes as `!`.

Verified with a forward-pass inspection:

```python
# All logits are NaN
logits = output.logits[0, -1]
# NaN count: 151936 out of 151936
```

And in the embeddings after `get_input_embeddings`:
```
embeds: NaN=0, Inf=2, max_abs=inf   # ← before fix
image features max abs: inf          # ← overflow inside vision tower
```

The overflow chain:
1. Vision tower (float16 weights) → 2 features exceed 65504 → `Inf`
2. `flat_x[positions] = flat_x[positions] + image_features` → embeds contain `Inf`
3. `attn_norm` (RMSNorm) block 0: `Inf / norm(Inf) = NaN`
4. NaN propagates to all 36 blocks and final logits
5. `argmax(NaN)` = 0 → token 0 → `!`

## Fix

In `mlx_vlm/models/molmo2/molmo2.py`, `get_input_embeddings`, guard and clip image features before merging:

```python
# After: image_features = self.vision_tower(pixel_values, token_pooling)
image_features = mx.where(mx.isnan(image_features), mx.zeros_like(image_features), image_features)
image_features = mx.clip(image_features, -65000, 65000)
```

A more robust fix would run the vision tower in bfloat16 (same exponent range as float32, no overflow risk), but that requires verifying the mixed-precision behaviour with the 4-bit weights.

## Environment

- `mlx-vlm`: 0.5.0
- `mlx`: 0.31.2
- Model: `mlx-community/Molmo2-8B-4bit`
- Hardware: Apple Silicon (macOS 25.4.0)
- Python: 3.12.7

### PR DESCRIPTION
**Pull request by lucasnewman**
Resolves https://github.com/Blaizzy/mlx-vlm/issues/1163

### CODE REVIEW

File: mlx_vlm/models/molmo2/molmo2.py [hunk 1]
```diff
@@ -9,6 +9,23 @@
...
+
+
+def image_feature_clip_limit(dtype) -> float:
+    dtype_max = float(mx.finfo(dtype).max)
+    if dtype == mx.float16:
+        return min(MAX_FLOAT16_IMAGE_FEATURE, dtype_max)
+    return dtype_max
+
+
+def sanitize_image_features(image_features: mx.array) -> mx.array:
```
Comment (Blaizzy) [inline mlx_vlm/models/molmo2/molmo2.py]: I would rename it to clip_image_features

**Review by Blaizzy — COMMENTED** [review]
LGTM, just one tiny change and we can merge

File: mlx_vlm/models/molmo2/molmo2.py (hunk 1 above)
Comment (lucasnewman) [reply mlx_vlm/models/molmo2/molmo2.py]: Good call

**Review by Blaizzy — APPROVED** [review]
🚀
